README.md

```diff
@@ -136,31 +136,32 @@ class TestAuthenticator extends OpenIdLoginAuthenticator
      */
     private $entityManager;
 
-    public function __construct(EntityManagerInterface $entityManager, SessionInterface $session, UrlGeneratorInterface $router)
+    public function __construct(EntityManagerInterface $entityManager, SessionInterface $session, UrlGeneratorInterface $router, OpenIdConfigurationProvider $provider)
     {
         $this->router = $router;
         $this->entityManager = $entityManager;
-        parent::__construct($session);
+        // Set leeway directly or configure via .env (must be positive)
+        parent::__construct($provider, $session, $leeway);
     }
 
     public function getUser($credentials, UserProviderInterface $userProvider)
     {
         $name = $credentials['name'];
         $email = $credentials['upn'];
 
-        //Check if user exists already - if not create a user
+        // Check if user exists already - if not create a user
         $user = $this->entityManager->getRepository(User::class)
             ->findOneBy(['email'=> $email]);
         if (null === $user) {
             // Create the new user
             $user = new User();
         }
-        // Update/set names here
+        // Update/set user properties
         $user->setName($name);
         $user->setEmail($email);
 
-        // persist and flush user to database
-        // If no change persist will recognize this
+        // Persist and flush user to database
+        // If no change, persist will recognize this
         $this->entityManager->persist($user);
```

Comment:
It's only necessary to call `persist()` on objects that are not handled by the `EntityManager`, should be moved inside the `if (null === $user) {` conditional

Reply:
Moved persist into conditional.